### PR TITLE
Add configuration for web api auth via Azure AD, and extend readme

### PR DESCRIPTION
**Pull request by daniellindemann**
### Motivation and Context

I want to authorize the calls from web app to web api via Azure AD (Entra ID). The application is prepared to use this feature but it's not fully implemented.  

This PR adds the required changes in the web app and the web api projects. It also contains configuration instructions.  
It resolves the issue #38 .

### Description

- adds `REACT_APP_AAD_API_SCOPE` to `Constants.ts` to request `access_as_user` scope via MSAL
- extends .env.example with the new introduced `REACT_APP_AAD_API_SCOPE` variable
- adds `Authorization:AzureAd:Audience` key to web api for request authorization checks
- adds a configuration instruction to `README.md`

### Contribution Checklist

- [x] The code builds clean without any errors or warnings
- [x] The PR follows the [Contribution Guidelines](https://github.com/microsoft/copilot-chat/blob/main/CONTRIBUTING.md) and the [pre-submission formatting script](https://github.com/microsoft/copilot-chat/blob/main/CONTRIBUTING.md#development-scripts) raises no violations
- [ ] All unit tests pass, and I have added new tests where possible
- [ ] I didn't break anyone :smile:

### CODE REVIEW

**Comment by daniellindemann** [comment]
@microsoft-github-policy-service agree company="abtis GmbH"

File: webapp/.env.example [hunk 1]
```diff
@@ -3,6 +3,9 @@
 REACT_APP_BACKEND_URI=https://localhost:40443/
 REACT_APP_AAD_AUTHORITY=https://login.microsoftonline.com/common
 REACT_APP_AAD_CLIENT_ID=
+# Authorization scopes to access webapi when using Azure AD authorization
+# example: api://12341234-1234-1234-1234-123412341234/access_as_user
```
Comment (teresaqhoang) [inline webapp/.env.example:7]: nit: point to the README section you added here

File: README.md [hunk 2]
```diff
@@ -127,6 +127,62 @@ First, let’s set up and verify the back-end API server is running.
...
+4. Add permissions to web app frontend to access web api as user
+   1. Open app registration for web app frontend
+
+   2. Go to *API Permissions*
+
+   3. Click *Add a permission*
+
+   4. Select the tab *My APIs*
+
+   5. Chooise the app registration representing the web api backend
```
Comment (glahaye) [inline README.md:170]: Typo

```suggestion
   5. Choose the app registration representing the web api backend
```

**Review by glahaye — DISMISSED** [review]

File: webapp/.env.example [hunk 3]
```diff
@@ -3,6 +3,9 @@
...
+REACT_APP_AAD_API_SCOPE=
```
Comment (dehoward) [inline webapp/.env.example:8]: move this below to the "Optional Variables" section?

File: README.md [hunk 4]
```diff
@@ -127,6 +127,62 @@ First, let’s set up and verify the back-end API server is running.
...
 
+## (Optional) Enable backend authorization via Azure AD
+
+1. Ensure you created the required application registration mentioned in [Start the WebApp FrontEnd application](#start-the-webapp-frontend-application)
+2. Create a second application registration to represent the web api
+   > For more details on creating an application registration, go [here](https://learn.microsoft.com/en-us/azure/active-directory/develop/quickstart-register-app).
+
+   1. Give the app registration a name
+
+   2. As *Supported account type* choose `Accounts in this organizational directory only (Single tenant)`
```
[the remark this answers is not in the document]
Comment (daniellindemann) [reply README.md]: No, it doesn't matter. I've changed it to match the previous configuration.

File: webapp/.env.example (hunk 1 above)
Comment (daniellindemann) [reply webapp/.env.example]: Added the readme reference and also added more detailed information for this env variable in the readme

File: webapp/.env.example (hunk 3 above)
Comment (daniellindemann) [reply webapp/.env.example]: Yep. The optional variables is a better place for the key. Thanks for the hint.

**Review by teresaqhoang — APPROVED** [review]

**Review by glahaye — APPROVED** [review]

**Review by crickman — APPROVED** [review]

**Pull request merged**